Remove platform dependent commands

Dabolus opened the pull request:
This PR replaces the `rm -rf` and the `mv` commands respectively with the `del-cli` and `move-cli` packages. It also makes some tweaks so that the `build:prpl-server` command can work seamlessly on any platform.

These tweaks will allow Windows users to use the starter kit without manually editing commands in the `package.json`. A future great improvement would be to use Gulp to run the tasks instead, so that we can have more flexibility (and also fix some security issues caused by `renamer` and `replace`). I will try to add some Gulp integration soon , but for now I preferred to fix this issue as soon as possible.

Fixes #95 

PS. Sorry for the dupe, I had some CLA problems with the other email 😅

Dabolus commented:
Don't know if it's the same case, but I had this same issue when I tried to run the build command after deleting the `server/` directory. Adding the `--mkdirp` flag to `move-cli` fixed the issue for me.

Dabolus commented:
About the `move-cli` popularity, strangely I couldn't find any popular package with a good CLI to move files and directories (`move-cli` was actually the most active and updated that i found). This gives me one more reason to just migrate to Gulp.

keanulee commented:
Hmm, still not working for me even with the flag. Gulp is probably the best way forward.

```
➜  pwa-starter-kit git:(pr/111) ✗ npx move-cli --mkdirp build server/build 
env: node\r: No such file or directory
➜  pwa-starter-kit git:(pr/111) ✗ npx move-cli --mkdirp build server/     
env: node\r: No such file or directory
➜  pwa-starter-kit git:(pr/111) ✗ npx move-cli --mkdirp              
env: node\r: No such file or directory
➜  pwa-starter-kit git:(pr/111) ✗
```

Dabolus commented:
Ok, I added some Gulp 4 integration and removed the no more used dependencies.
The build command should also **theorically** be faster, as with Gulp `build:static` and `build:prpl-server` are executed in parallel. The build process is basically the same as the one done by the CLI, but with some edits to fit it with our needs (e.g. renaming `node_modules` is handled at build time)

keanulee commented on .gitignore:
```diff
@@ -3,3 +3,4 @@ node_modules/
 test/integration/screenshots-current
 yarn.lock
 .DS_Store
+.idea
```
Don't think this is needed.

keanulee commented on gulpfile.js:
```diff
@@ -0,0 +1,205 @@
+    })),
+  };
+  const project = new PolymerProject(autoBasePathedConfig);
+  return Promise.all(autoBasePathedConfig.builds.map((buildConfig) =>
```
Need to output polymer.json so that it can be used by `npm run serve:prpl-server` - similar to https://github.com/Polymer/tools/blob/0cdecd03e7e3709f84fa4f5a52443c19988112be/packages/cli/src/commands/build.ts#L214

notwaldorf commented:
on it!

notwaldorf commented on gulpfile.js:
```diff
@@ -0,0 +1,214 @@
+const gulp = require('gulp');
```
Needs license

notwaldorf commented:
Doesn't work for me (i did a clean `npm install`) on Node 8.2.1 and npm 6.0.0.
<img width="958" alt="screen shot 2018-05-18 at 12 02 51 pm" src="https://user-images.githubusercontent.com/1369170/40253223-dc961e00-5a93-11e8-83f7-9be116e88463.png">

notwaldorf commented on package.json:
```diff
@@ -8,9 +8,9 @@
-    "build:static": "polymer build",
+    "build": "gulp build",
+    "build:prpl-server": "gulp clean:prpl-server && gulp build:prpl-server",
+    "build:static": "gulp clean:static && gulp build:static",
```
why does `build:static` need to change? it doesn't have to do any work, so it doesn't need to be built by `gulp`

keanulee replied:
I guess technically `build:static` is still functionally equivalent to `polymer build`

notwaldorf commented on package.json:
```diff
@@ -8,9 +8,9 @@
-    "build": "npm run build:prpl-server && npm run build:static",
-    "build:prpl-server": "polymer build --auto-base-path && rm -rf server/build && mv build server/ && renamer --find node_modules --replace node_assets 'server/build/**' && replace node_modules node_assets server/build -r -s",
-    "build:static": "polymer build",
+    "build": "gulp build",
```
why does this have to change?

notwaldorf commented on gulpfile.js:
```diff
@@ -0,0 +1,214 @@
+  const buildDirectory = path.join(options.outputDir, buildName);
+  const logPrefix = options.logPrefix || '';
+
+  logger.info(`${logPrefix}(${buildName}) Preparing...`);
```
why can't we just `console.log`?

keanulee replied:
Code from polymer-cli. Does nice formatting stuff, I think.

notwaldorf replied:
i guess my biggest comment is why do we have to redo all the logic `polymer build` does instead of...exec-ing it.

notwaldorf commented on gulpfile.js:
```diff
@@ -0,0 +1,214 @@
+const rename = require('gulp-rename');
+const logger = require('gulplog');
+const path = require('path');
+const fs = require('mz/fs');
```
what's wrong with the builtin `fs` package?

keanulee replied:
Code from polymer-cli

notwaldorf commented on gulpfile.js:
```diff
@@ -0,0 +1,214 @@
+const fs = require('mz/fs');
+const del = require('del');
+const mergeStream = require('merge-stream');
+const {forkStream, PolymerProject, addServiceWorker, HtmlSplitter, getOptimizeStreams} = require('polymer-build');
```
since we already have the `polymer cli` installed, can't we just.......`exec` `polymer build`?

keanulee replied:
Most of the code is from polymer-cli, but having it here means it can run `build:static` and `build:prpl-server` in parallel (since they won't share the same /build output directory).

Dabolus commented:
I moved all the build process to gulp not only to allow parallel building, but also to avoid making the additional post-processing steps after the build. By the way, I agree on the fact that this is pretty much like reinventing the wheel and it can cause a lot of confusion. I'll be back with the edits in a minute

keanulee commented:
@frankiefu @notwaldorf PTAL

frankiefu commented on package.json:
```diff
@@ -8,8 +8,8 @@
     "start": "polymer serve",
-    "build": "npm run build:prpl-server && npm run build:static",
-    "build:prpl-server": "polymer build --auto-base-path && rm -rf server/build && mv build server/ && renamer --find node_modules --replace node_assets 'server/build/**' && replace node_modules node_assets server/build -r -s",
+    "build": "gulp build",
```
I'd suggest to keep `build` like it was before (`"build": "npm run build:prpl-server && npm run build:static"`) and the `gulpfile.js` will be just for `prpl-server` stuffs.

Btw, thanks for working on the PR @Dabolus!

frankiefu commented on gulpfile.js:
```diff
@@ -0,0 +1,66 @@
+gulp.task('build:static', (cb) =>
+  // Build the project using Polymer CLI
+  exec('polymer build', cb));
+
```
Would you mind removing this unused gulp task too? Thanks!

Dabolus replied:
Sorry, forgot about that one!

frankiefu commented:
@Dabolus Thanks for all the changes. One last thing, have you ran the script on Windows to make sure the PR really fixes https://github.com/Polymer/pwa-starter-kit/issues/95?

Dabolus commented:
@frankiefu yep, I tried it on my old laptop (Windows 10, Node v10.0.0, npm v6.0.0). The build was successful and the output files in the server `server/build/` directory were correctly post-processed

frankiefu commented:
@Dabolus That's awesome! Thanks!
